hw/ip/entropy_src/doc/_index.md

```diff
@@ -10,83 +10,145 @@ This module conforms to the [Comportable guideline for peripheral functionality.
+
+These raw noise bits are subjected to a sequence of health-checks to screen the raw signals for statistical defects which would cause any significant deviations from ideal entropy at the output of the conditioning block.
+These tests include:
+- The Repetition Count test, which screens for stuck-bits, or a complete failure of the physical noise source,
+- The Adaptive Proportion test, which screens for statistical bias in the number of 1's or 0's output by the noise source,
+- The "Bucket Test", which looks for correlations between the individual noise channels that the external noise source concatenates together to produce the raw noise sequence,
+- The "Markov Test", which looks for unexpected first-order temporal correlations between bits output by the individual noise channels,
+- The "Mailbox Test", in which raw-noise data is transferred to firmware in contiguous 2kbit-blocks, so that firmware can perform custom tests, and signal a failure through the same path as the other tests, and
+- Optional Vendor Specific tests, which allow silicon creators to extend the health checks by adding a top-level block external to this IP.
+
+The Repetition Count and Adaptive Proportion test are specifically recommended by SP 800-90B, and are implemented in accordance with those recommendations.
+In FIPS/CC-compliance mode, all checks except the Repetition Count test are performed on fixed window of data, typically consisting of 1024 bits each.
+Per the definition in SP 800-90B, the Repetition Count test does not operate on a fixed window.
+The repetition count test fails if any sequence of bits continuously asserts the same value for too many samples, as determined by the programmable threshold, regardless of whether that sequence crosses any window boundaries.
+The thresholds for these tests should be chosen to achieve a low false-positive rate (&alpha;) given a conservative estimate of the manufacturing tolerances of the physical noise source.
+The combined choice of threshold and window size then determine the false-negative rate (&beta;), or the probability of missing statistical defects at any particular magnitude.
+
+When the IP is disabled by clearing the `ENABLE` bit in {{< regref "CONF" >}}, all heath checks are disabled and all counters internal to the health checks are reset.
 
-Once the threshold has been reached for collected entropy bits, the firmware has two options of notification: an interrupt, or polling.
-The `es_entropy_valid` bit will be set when the {{< regref "ES_THRESH" >}} register value is equal or less than the number of elements in the TLUL FIFO.
-Optionally, an interrupt may be a better use of CPU cycles.
+In order to compensate for the fact our tests (like *all* realistic statistical tests) have finite resolution for detecting defects, we conservatively use 1024 bits of raw digital noise to construct each 384 bit key.
```

Comment:
the word "key" here seems weird to me.  It's really "conditioned entropy sample" right?  (Feel free to wordsmith for NIST-appropriate terminology if need be)